### PR TITLE
Implementing a reverse operator

### PR DESCRIPTION
**Pull request by lepiolet**
This PR implements 2 new operators, having in total 3:
 - sdp-operator :: syncs entities in a k8s with an appgate system  
 - sdp-operator in reverse mode :: syncs an appgate system with k8s
 - git-operator (github only for now) :: syncs k8s with git (creating pull request with the changes in the state)

This allows us to migrate entities between appgate systems:

1. appgate system A where we can use the UI to edit entities
2. the `sdp-operator` in reverse mode will sync that state into k8s
3. the `git-operator` create pull requests in github with the entities created by the `sdp-operator` running in reverse mode.
4. Once an admin merges the pull requests, argocd will add those into k8s
5. A normal `sdp-operator` will sync an appgate system B with the entities added by argocd.

Pull request look like this:

image

The `git-operator` right now keeps accumulating commits with changes into one pull request until this pull request is merged. Then it will create a new branch and a new pull request from it.

In case the `git-operator` is restarted it will reuse the latest pull request opened (with `sdp-operator`) tag.

### CODE REVIEW

File: appgate/appgate.py [hunk 1]
```diff
@@ -401,40 +312,67 @@ async def operator(
...
+                if not ctx.dry_run_mode and not ctx.reverse_mode:
+                    entity_clients = generate_api_spec_clients(
+                        api_spec=ctx.api_spec, appgate_client=appgate_client
+                    )
+                if not ctx.dry_run_mode and ctx.reverse_mode:
+                    entity_clients = generate_k8s_clients(
+                        api_spec=ctx.api_spec,
+                        namespace=ctx.namespace,
+                        k8s_api=get_crds(),
+                    )
```
Comment (mandopaloooza) [inline appgate/appgate.py:329]: Interesting way of implementing the dry run.  Took me a bit to understand but it makes sense. If the entity_clients is None, `plan_apply` doesn't run any of the entity_client's action.

**Review by mandopaloooza — APPROVED** [review]
Looks good to me. Really like the improvements to the PR and git commit messaging 👍

File: appgate/syncer/operator.py [hunk 2]
```diff
@@ -0,0 +1,285 @@
...
+        dry_run=dry_run_mode,
+        git_vendor=ensure_env(GIT_VENDOR_ENV),
+        git_repository=ensure_env(GIT_REPOSITORY_ENV),
+        git_base_branch=ensure_env(GIT_BASE_BRANCH_ENV),
+        log_level=os.environ.get(APPGATE_LOG_LEVEL, "info"),
+        git_repository_fork=os.environ.get(GIT_REPOSITORY_FORK_ENV),
+        main_branch=os.environ.get(
+            GIT_REPOSITORY_MAIN_BRANCH_ENV, GIT_REPOSITORY_MAIN_BRANCH
+        ),
+    )
```
Comment (mandopaloooza) [inline appgate/syncer/operator.py:93]: What is the difference between `git_base_branch` and `main_branch`?

Comment (lepiolet) [reply appgate/syncer/operator.py:93]: None, I think that's just a duplication. I will remove one.

**Pull request merged**